modules/scala-support/src/main/scala/dev/guardrail/generators/scala/circe/CirceRefinedProtocolGenerator.scala

```diff
@@ -28,6 +28,40 @@ object CirceRefinedProtocolGenerator {
   def applyValidations(className: String, tpe: Type, prop: Tracker[Schema[_]]): Target[Type] = {
     import scala.meta._
     tpe match {
+      case Type.Apply(t"Vector", inner) =>
+        val containerElementSchema: Tracker[Option[Schema[_]]] = prop.downField("items", _.getItems)
+        val innerType: Target[Type] = containerElementSchema.fold(
+          Target.fromOption(inner.headOption, RuntimeFailure(s"Unexpected state - empty 'items': ${containerElementSchema.showHistory}"))
+        ) { tracker =>
+          Target
```

Comment:
there is too much nesting and the error propagation doesn't look right, how can one improve on this?